File: app/testdata/simple.py
# A simple Python file for testing

def hello(name):
    """Greet someone."""
    return f"Hello, {name}"


def calculate(x, y, operation):
    if operation == "add":
        return x + y
    elif operation == "sub":
        return x - y
    elif operation == "mul":
        return x * y
    elif operation == "div":
        if y == 0:
            raise ValueError("Cannot divide by zero")
        return x / y
    else:
        raise ValueError(f"Unknown operation: {operation}")


def process_items(items):
    results = []
    for item in items:
        if item is None:
            continue
        if isinstance(item, str) and len(item) > 0:
            results.append(item.upper())
        elif isinstance(item, int):
            results.append(item * 2)
    return results

File: app/testdata/simple.ts
// A simple TypeScript file for testing

function greet(name: string): string {
  return `Hello, ${name}`;
}

function calculate(x: number, y: number, op: string): number {
  if (op === "add") {
    return x + y;
  } else if (op === "sub") {
    return x - y;
  } else if (op === "mul") {
    return x * y;
  } else if (op === "div") {
    if (y === 0) {
      throw new Error("Division by zero");
    }
    return x / y;
  }
  throw new Error(`Unknown op: ${op}`);
}

const processItems = (items: (string | number | null)[]): (string | number)[] => {
  const results: (string | number)[] = [];
  for (const item of items) {
    if (item === null) {
      continue;
    }
    if (typeof item === "string" && item.length > 0) {
      results.push(item.toUpperCase());
    } else if (typeof item === "number") {
      results.push(item * 2);
    }
  }
  return results;
};
